### ISSUE TITLE
Peer Review Group C

### ISSUE THREAD

**Issue by NicoHenzel** [Raalit00/Statistic_Project#1]
### Peer review team

- Peer review by: Group C
- Names of team members that participated in this review: Burcin Aksel & Nico Henzel

### Goal of the project

- Predict how many yards will be gained on a play by play basis based on variables that are known before a play starts.

### Data

- The data is provided by the NFL Gen Stats website for the 2022 season up to december 26th. 
 The data contains variables for football games.
 A subset is selected for variables that are known before a play.
 It is granular and large enough to train a machine learning model. 

### Approach, tools and methods

- Data is imported and corrected.
 A subset of the data is made by only selecting variables that are known before a play. 
 The variables are explained within a data dictionary, and some data exploration is made.
 The relation and correlation from the subset with the response variable is displayed.

- The tools used are well known data manipulation and machine learning libraries such as
pandas, numpy, altair, matplotlib and sklearn

- A pairplot matrix and correlation matrix is made to show relation and correlation of variables.
 The data is split into training and test set for the models.
 The regression is modelled by using Linear, Lasso and KNN Regression algorithms.
 These are then compared to a model that always predicts the mean of yards gained (similar to a naive predictor that sets a baseline).

### Lack of clarity

- We could have used a more detailed explanation on the rules of the game in order to understand some variables better.
For example when you explain the correlation between ydstgo and yardline_100.

- The research question could be clearer. 
 Do you only want to identify the importance of variables for predicting yards gained?
 Or do you want to go further and build a model that is able to predict which team will gain how many yards?
 Maybe you could ask: "Determine the factors that allow a prediction of yards gained for a team in a football match on play by play basis."

- The descriptive statistics and exploratory data analysis could be explained in more detail:
 What do the numbers for the descriptive statistc say? 
 Why is the distribution like you expected?

### Possible improvements

- The text right before the Data Dictionary seems to be the same information displayed in the table. 
 Is that necessary?

- You should show the original dataset with its features so it is clear which variables are known before a play.
 This way the reader can understand why you chose your variables for df_small.

- Under data preparation we encountered an error with the code to generate the dummies:
dummies = pd.get_dummies(X[list_categorial])
dummies.info()
X = pd.concat([X[list_numeric], dummies], axis=1)
X.info()
We needed to add slicing in order to disregard "yards_gained" since it is not found in X:
X = pd.concat([X[list_numeric[1:]], dummies], axis=1)

 You could also explain why you need dummy variables in the first place.

- For all visualizations:
 Use titles and proper variable names (i.e. without "_") like you did for the lasso regression model.

- You don't explain what the visuals for the lasso regression model mean.

- Under Conclusion you wrote that KNN was able to beat the mean value model. 
 Is that a typo because the error values of KNN are higher than those of the mean value model.

- You could use R² as an additional metric to evaluate your mode performance and underline your interpretation.

- In your models, you could try to include more / other variables and see if the performance changes. 
 This could maybe lead to a different conclusion.
 You can even use the whole original dataset and do forward / backward selection: 
 https://as-ws22.github.io/website/code/56-wrapper.html

- Nice to have: visualization of the regression line fitted by the models.

### Presentation

- The most interesting part is how the research question can be answered.
 Is something similar already used by NFL Teams and if yes, what kind of data do they use?
 
- Also what could be done in order to improve the models further, what the next steps for a follow-up project could be.

### Organization

- The only improvements for code organizations is to consistently use comments for every block of code. 
 Other than that the code is clean and easy to read.

- The data is not clearly located within the data folder.
The cleaned dataset is found within the raw folder together with the original dataset.
Is the data_cleaned_2021.csv the original dataset?

- There is no model being saved for the Lasso Regression.
The linear regression and KNN Regression Model file have the same name.

### Further comments

- Do you have additional content for the appendix?
 Because there is none in the notebook yet.

- If you want to put the icing on the cake for interactive graphs you could use tooltips:
alt.Chart(df_tmp).mark_bar().encode(
    x=alt.X("yards_gained"),
    y=alt.Y('count()'),
    tooltip = ['yards_gained']
).interactive()
